src/lib.rs

```diff
@@ -236,7 +236,12 @@ pub fn narrowable_rboehm(args: TokenStream, input: TokenStream) -> TokenStream {
                     ::std::ptr::write(vtablep as *mut usize, vtable);
 
                     init(objp as *mut U);
-                    gc.assume_init()
+                    let mut init = gc.assume_init();
+
+                    if !::rboehm::gc::needs_finalizer::<U>() {
```

Comment:
Will this turn into a compile-time constant in most cases?

Reply:
Yes as long as Rust manages to inline this ( I don't see why it won't ). The underlying intrinsics are both const expressions.

Reply:
Just checking :)